lib/nrf_modem_lib/nrf_modem_os.c

```diff
@@ -561,7 +620,8 @@ void nrf_modem_os_init(void)
 		    CONFIG_NRF_MODEM_LIB_SHMEM_TX_SIZE);
 
 #if defined(CONFIG_NRF_MODEM_LIB_SHM_TX_DUMP_PERIODIC) || \
-	defined(CONFIG_NRF_MODEM_LIB_HEAP_DUMP_PERIODIC)
+	defined(CONFIG_NRF_MODEM_LIB_HEAP_DUMP_PERIODIC) || \
+	defined(CONFIG_NRF_MODEM_LIB_TRACE_HEAP_DUMP_PERIODIC)
 	k_work_queue_start(&modem_diag_worqk, work_q_stack_area,
```

Comment:
Looks like `modem_diag_worqk` is not used at all, because the periodic dumps are done in the system work queue?